### DISCUSSION TITLE
Debugging in vscode

### DISCUSSION THREAD

**Discussion by alexander-kolarov-tiebreak** [vercel/turborepo#8401] [Help]
### Summary

I cant run debugger in VSCode.

Can you show some `launch.json` that would hit breakpoints in `apps` client-side and server-side. Especially in server-side.

I run the whole app from the root and want to debug `apps/app1` for example.

The only thing that worked(partially) was to run debug configuration directly in `apps/app1`. Thus the breakpoint hits but the root task was not ran so the project did not work correctly.

**Comment by anthonyshew** [comment]
Is this a Turborepo or Turbopack question?

**Reply by alexander-kolarov-tiebreak** [reply]
> Is this a Turborepo or Turbopack question?

The project is using turbo for bundling and pnpm workspaces. It has the same structure as https://github.com/vercel/turbo/tree/main/examples/basic.

**Reply by notaphplover** [reply]
@anthonyshew it's related to turborepo. Probably related to https://github.com/vercel/turbo/discussions/8388

**Comment by goshantmeher** [comment]
for me this works for nestJs, nextJs
launch.json:

```
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Debug API",
      "type": "node",
      "request": "launch",
      "runtimeExecutable": "pnpm",
      "runtimeArgs": ["run", "start:debug"],
      "cwd": "${workspaceFolder}/apps/api"
    },
    {
      "name": "Debug Web",
      "type": "node-terminal",
      "request": "launch",
      "command": "pnpm run dev",
      "cwd": "${workspaceFolder}/apps/web"
    }
  ],
  "compounds": [
    {
      "name": "Debug All",
      "configurations": ["Debug API", "Debug Web"]
    }
  ]
}
```
